### ISSUE TITLE
should span start-time attributes to be set on started span

### ISSUE THREAD

**Issue by lmolkova** [open-telemetry/opentelemetry-java#3421]
**Describe the bug**

I assume if I provide attributes to span start, they should be populated on the span, but it's not the case

```java
var span = openTelemetry.getTracer("foo")
	.spanBuilder("bar")
	.setAttribute("k", "v")
	.startSpan();
span.end();

System.out.println(((ReadableSpan)span).toSpanData().getAttributes().asMap().get("k"));
```

output is `null`. 

Spec does not explicitly explain that, but following

> Whenever possible, users SHOULD set any already known attributes at span creation instead of calling SetAttribute later.

probably means that SDK should set them on the span.

**What version and what artifacts are you using?**
Artifacts:  `opentelemetry-sdk`
Version: `1.4.0`
How did you reference these artifacts? (excerpt from your `build.gradle`, `pom.xml`, etc): N/A

**Environment**
Compiler: N/A
OS: N/A
Runtime N/A
OS (if different from OS compiled on): (e.g., "Windows Server 2019")

**Additional context**

Looking [into the code](https://github.com/open-telemetry/opentelemetry-java/blob/daef9c147daf836ef84c05f0d139e957c70b37bc/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java#L187), it seems those attributes only used to make sampling decisions but never populated on span.

**Comment by jkwatson** [comment]
Note: https://github.com/open-telemetry/opentelemetry-java/blob/daef9c147daf836ef84c05f0d139e957c70b37bc/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java#L217-L231

**Comment by jkwatson** [comment]
But, more importantly, you're trying to get a string "k" as the key to the Attributes, but the keys to the Map are AttributeKey<?>, not String.

**Comment by lmolkova** [comment]
thanks, my bad!  This is not very user-friendly API, though :)

**Comment by jkwatson** [comment]
> thanks, my bad! This is not very user-friendly API, though :)

You're trying to do things with the SDK that were not intended. the API gives you a write-only span for a reason. :)

**Comment by lmolkova** [comment]
We do all the crazy things in tests for the isntrumentation

**Comment by jkwatson** [comment]
> We do all the crazy things in tests for the isntrumentation

of that, I'm 100% sure. :)

but, seriously, we spent a lot of time designing the Attribute API to make it type-safe for writing, since that is the primary usage. We assume those who are using the SDK are more willing to put up with a few warts at export time. It was a very hard to optimize for all possible use-cases, so we chose the API user over the SDK user.
